### DISCUSSION TITLE
Detect if the device is using VoIP or Virtual Number

### DISCUSSION THREAD

**Discussion by QaunainM** [camaraproject/DeviceStatus#147] [Ideas]
Hi all,  some customers mentioned they want to detect if a phone number is a VoIP or Virtual Number.

I don't see this type of data point inside DeviceStatus or in the brief for ConnectivityInsights

Have you heard about a Camara API way to detect if a user is on a VOIP network, or could this be added as a data point withing DeviceStatus?

**Comment by QaunainM** [comment]
@NoelWirzius @akoshunyadi @bigludo7, I wonder what your thoughts on the above are, thanks

**Comment by eric-murray** [comment]
@QaunainM
Do you have a particular definition of "VOIP network" in mind?

I assume this excludes 3rd party (OTT) VoIP services such as WhatsApp. Is it enough that the access network used for the voice service is packet switched only, or must the voice service be supported by a specific type of core (e.g. IMS)?

Also, is the requirement to determine whether the device is in an active VoIP session, or simply that, if the device were to (hypothetically) make or receive a voice call at that instant in time, would the call be connected via a VoIP (packet switched) or circuit switched network?

**Comment by QaunainM** [comment]
Hi @eric-murray , 

The use case would be to detect if the number is a VOIP Number or Virtual Number, i.e. a number that someone could purchase online without having to go through a traditional Telco.  

Examples of problems it solves are:

- Stops OTPs and SMS being sent to virtual numbers  
- Lets companies know that people are signing up with (or changing their number to) virtual numbers instead of 'real numbers' so they can block or limit the activity for those user types

 Another use case is mentioned below, but I don't think detecting Virtual numbers will solve it as machines could also have Telco provided physical or eSIMS

- Customers need to know if the number is a human or machine

**Comment by eric-murray** [comment]
So you mean virtual numbers issued by the likes of Skype or Twilio? For network operators, these are OTT services using our data network.

**Comment by QaunainM** [comment]
Yeah exactly those types of numbers. I wonder if there is a way for the response to display that more descriptively, i.e. Virtual Number: Yes / No

**Comment by QaunainM** [comment]
Hi all, hope everyone is doing good. Just circling back on this topic, do we think that Device Status (reachability) in the future would be able to return if the number is a VoIP number like the ones @eric-murray described above. 

This seems to be a use case I've heard interest about.
